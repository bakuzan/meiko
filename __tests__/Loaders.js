import React from 'react';

import {
  LoadingBouncer,
  LoadingSpinner,
  LoadableContent,
  SimpleLoading,
  Loading
} from '../lib';

jest.useFakeTimers();

describe('LoadingBouncer', function() {
  it('should render with minimum props', function() {
    const component = shallow(<LoadingBouncer />);

    expect(component.is('.loading-bouncer')).toBeTruthy();
    expect(component).toMatchSnapshot();
  });
});

describe('LoadingSpinner', function() {
  it('should render with minimum props', function() {
    const component = shallow(<LoadingSpinner />);

    expect(component.is('.loader')).toBeTruthy();
    expect(component).toMatchSnapshot();
  });
});

describe('SimpleLoading', function() {
  it('should render with minimum props', function() {
    const component = shallow(<SimpleLoading />);

    expect(component.html()).toBeNull();
  });

  it('should render', function() {
    const component = shallow(<SimpleLoading pastDelay={true} />);

    expect(component.is('.loading-bouncer')).toBeTruthy();
    expect(component).toMatchSnapshot();
  });
});

describe('Loading', function() {
  it('should render with minimum props', function() {
    const component = shallow(<Loading />);

    expect(component.html()).toBeNull();
  });

  it('should render', function() {
    const component = shallow(<Loading pastDelay={true} />);

    expect(component.is('.loader')).toBeTruthy();
    expect(component).toMatchSnapshot();
  });

  it('should render error message', function() {
    const component = shallow(<Loading error={true} />);

    expect(component.text()).toEqual(
      'An Error was encountered loading the page!'
    );
    expect(component).toMatchSnapshot();
  });

  it('should render timeout message', function() {
    const component = shallow(<Loading timedOut={true} />);

    expect(component.text()).toEqual('The request has timed out!');
    expect(component).toMatchSnapshot();
  });
});

describe('LoadableContent', function() {
  it('should render with minimum props', function() {
    const component = shallow(
      <LoadableContent>
        <div id="jest" />
      </LoadableContent>
    );

    expect(component.find('#jest').exists()).toBe(true);
    expect(component).toMatchSnapshot();
  });

  it('should render spinner after timeout', function() {
    const component = shallow(
      <LoadableContent isFetching={false}>
        <div id="jest" />
      </LoadableContent>
    );

    expect(component.state('pastDelay')).toBe(false);

    component.setProps({ isFetching: true });
    jest.runAllTimers();

    expect(component.state('pastDelay')).toBe(true);
    expect(component.is('.loader')).toBeTruthy();
    expect(component).toMatchSnapshot();
  });
});
