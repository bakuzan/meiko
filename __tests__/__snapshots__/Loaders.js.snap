// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadableContent should render with minimum props 1`] = `
<div
  id="jest"
/>
`;

exports[`LoadingBouncer should render with minimum props 1`] = `
<div
  className="loading-bouncer  mko_flo780-bouncer"
>
  <div
    className="loading-bouncer__orb  mko_flo780-orb"
    key="0"
  />
  <div
    className="loading-bouncer__orb  mko_flo780-orb"
    key="1"
  />
  <div
    className="loading-bouncer__orb  mko_flo780-orb"
    key="2"
  />
</div>
`;

exports[`LoadingSpinner should render with minimum props 1`] = `
<div
  className="loader  mko_zz2xb6-loader"
>
  <svg
    className="loader__circular  mko_zz2xb6-circular"
    viewBox="25 25 50 50"
  >
    <circle
      className="loader__path  mko_zz2xb6-path"
      cx="50"
      cy="50"
      fill="none"
      r="20"
      strokeMiterlimit="10"
      strokeWidth="2"
    />
  </svg>
</div>
`;
